### PR TITLE
PORTALS-2355: pin a discussion thread

### PR DESCRIPTION
**Pull request by yjyoo773**

### CODE REVIEW

File: src/lib/containers/discussion_forum/DiscussionThread.tsx [hunk 1]
```diff
@@ -159,9 +159,18 @@ export function DiscussionThread(props: DiscussionThreadProps) {
...
-          <button onClick={() => setShowDeleteModal(true)}>
-            <IconSvg icon="delete" label="Delete thread" />
-          </button>
+          <>
+            <button onClick={() => setShowDeleteModal(true)}>
+              <IconSvg icon="delete" label="Delete thread" />
+            </button>
+            <button onClick={() => togglePin()}>
+              {threadData?.isPinned ? (
+                <IconSvg icon="block" label="Unpin thread" />
```
Comment (yjyoo773) [inline src/lib/containers/discussion_forum/DiscussionThread.tsx:168]: Recommend a different icon for unpinning?

**Review by nickgros — CHANGES_REQUESTED** [review]

File: src/lib/utils/hooks/SynapseAPI/forum/useForum.ts [hunk 2]
```diff
@@ -205,30 +207,43 @@ export function useCreateThread(
...
     {
       ...options,
-      onSuccess: async (newThread, variables, ctx) => {
-        await queryClient.invalidateQueries(['forumthread', newThread.forumId])
+      onSuccess: async (threadBundle, newThreadRequest, ctx) => {
+        await queryClient.invalidateQueries([
+          'forumthread',
+          threadBundle.forumId,
+        ])
+        await queryClient.invalidateQueries(['thread', threadBundle.id])
```
Comment (nickgros) [inline src/lib/utils/hooks/SynapseAPI/forum/useForum.ts:215]: Is this line necessary? The thread is new so we shouldn't have any information about its id in the cache.
```suggestion
```

File: src/lib/utils/hooks/SynapseAPI/forum/useForum.ts [hunk 3]
```diff
@@ -205,30 +207,43 @@ export function useCreateThread(
...
+      SynapseClient.deleteThread(accessToken, threadBundle.id),
     {
       ...options,
-      onSuccess: async (updatedThread, threadId, ctx) => {
-        await queryClient.invalidateQueries(['thread', threadId])
+      onSuccess: async (updatedThread, threadBundle, ctx) => {
+        await queryClient.invalidateQueries([
+          'forumthread',
+          threadBundle.forumId,
+        ])
```
Comment (nickgros) [inline src/lib/utils/hooks/SynapseAPI/forum/useForum.ts:243]: Can you also add this `'forumthread' invalidation to `useUpdateThreadTitle`?

**Review by nickgros — CHANGES_REQUESTED** [review]

File: src/lib/containers/discussion_forum/DiscussionReply.tsx [hunk 4]
```diff
@@ -52,51 +50,48 @@ export const DiscussionReply: React.FC<DiscussionReplyProps> = ({
...
+                avatarSize="MEDIUM"
+                showCardOnHover={true}
+                size={SMALL_USER_CARD}
+                ownerId={reply.createdBy}
+              />
+              <div className="message-body">
+                <MarkdownSynapse
+                  markdown={message}
+                  objectType={ObjectType.REPLY}
+                />
```
Comment (nickgros) [inline src/lib/containers/discussion_forum/DiscussionReply.tsx:72]: Probably just need the skeleton here since you have all of the other reply fields

**Review by nickgros — APPROVED** [review]
One small suggestion

File: src/lib/containers/discussion_forum/DiscussionReply.tsx (hunk 4 above)
Comment (yjyoo773) [reply src/lib/containers/discussion_forum/DiscussionReply.tsx:72]: When updating, the reply gets rendered. So its original position should be better.

**Pull request merged**